src/neo/Ledger/Blockchain.cs

```diff
@@ -498,13 +499,17 @@ private void Persist(Block block)
 
                     snapshot.Transactions.Add(tx.Hash, state);
```

Comment:
You added a new record to the snapshot after `Clone()`. It may be safe now, but I am not sure if it will cause problems in the future. For example, cache non-existent records.

Reply:
That's right. Maybe adding a todo comment to warn developer for future modification can help.

Reply:
I think it conflicts with the cache to the non-existent keys. Maybe we should test which one is better first?

Reply:
Have updated the writing logic here & should be able to avoid such occasions:)